src/llfs/ioring_file.test.cpp

```diff
@@ -0,0 +1,61 @@
+//#=##=##=#==#=#==#===#+==#+==========+==+=+=+=+=+=++=+++=+++++=-++++=-+++++++++++
+//
+// Part of the LLFS Project, under Apache License v2.0.
+// See https://www.apache.org/licenses/LICENSE-2.0 for license information.
+// SPDX short identifier: Apache-2.0
+//
+//+++++++++++-+-+--+----- --- -- -  -  -   -
+
+#include <llfs/ioring_file.hpp>
+//
+#include <llfs/ioring_file.hpp>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <llfs/filesystem.hpp>
+#include <llfs/track_fds.hpp>
+
+namespace {
+
+using namespace llfs::int_types;
+
+using llfs::Status;
+using llfs::StatusOr;
+
+TEST(IoringFileTest, FdLeakTest)
+{
+  bool saved = llfs::set_fd_tracking_enabled(true);
+  auto on_scope_exit = batt::finally([&] {
+    llfs::set_fd_tracking_enabled(saved);
+  });
+
+  StatusOr<llfs::ScopedIoRing> scoped_io_ring =
+      llfs::ScopedIoRing::make_new(llfs::MaxQueueDepth{32}, llfs::ThreadPoolSize{1});
+
+  ASSERT_TRUE(scoped_io_ring.ok()) << BATT_INSPECT(scoped_io_ring.status());
```

Comment:
Consider using `BATT_CHECK_OK` instead.

Reply:
Well, in this case I think its better to have the gtest binary finish with a record of what failed than for it to panic.  But this is a common enough idiom at this point that it would be nice for us to have our own gtest extension to handle it (i.e., `EXPECT_OK` and `ASSERT_OK`).

Created https://gitlab.com/batteriescpp/batteries/-/issues/19